tests/networkpolicy_test.go

```diff
@@ -223,17 +320,44 @@ func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
 	}, 5*time.Second, 1*time.Second).ShouldNot(Succeed())
 }
 
-func waitForNetworkPolicyDeletion(policy *v1network.NetworkPolicy) {
+func createNetworkPolicy(policy *networkv1.NetworkPolicy) {
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	By(fmt.Sprintf("Create %s networkpolicy", policy.Name))
+	_, err = virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Create(policy)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed creating network policy %s/%s", policy.Namespace, policy.Name))
+}
+
+func waitForNetworkPolicyDeletion(policy *networkv1.NetworkPolicy) {
 	if policy == nil {
 		return
 	}
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &v13.DeleteOptions{})).To(Succeed())
+	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &metav1.DeleteOptions{})).To(Succeed())
 	EventuallyWithOffset(1, func() error {
-		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, v13.GetOptions{})
+		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, metav1.GetOptions{})
 		return err
-	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())))
+	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())))
+}
+
+func PingHTTPEventually(vmiFrom, vmiTo *v1.VirtualMachineInstance, port int) AsyncAssertion {
+	const curlCheckCmd = "curl --head %s --connect-timeout 5\n"
+	return Eventually(func() error {
+		for _, ip := range vmiTo.Status.Interfaces[0].IPs {
+			if netutils.IsIPv6String(ip) {
```

Comment:
btw, i read somewhere that even ipv4 can be encapsulated with "[" if you want to try

Reply:
Not working

```bash
[ellorent@localhost cluster-network-addons-operator]$ curl -k https://[216.58.201.164]:443
curl: (3) URL using bad/illegal format or missing URL
[ellorent@localhost cluster-network-addons-operator]$ curl -k https://[216.58.201.164]
curl: (3) URL using bad/illegal format or missing URL
[ellorent@localhost cluster-network-addons-operator]$ curl -k https://216.58.201.164
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>
```

Reply:
Thanks for checking